docs/development.md

```diff
@@ -35,3 +35,20 @@ Since this is very unlikely and most times an abs path is not strictly necessary
 "Should" functions can be used anywhere,
 but care should be taken not to overuse them when a fallback value is not sufficient.
 "Must" variants should be favored in testing code, since they will more obviously fail tests.
+
+## Cross-Platform Scripting
+
+We are writing bash scripts with the intention of running them in a POSIX environment. However, differences between operating systems, particularly between Linux and MacOS, can still affect a script's behavior. This section highlights key considerations to ensure your scripts run consistently across platforms.
```

Comment:
I tend to put sentences/clauses on lines by themselves so that diffs come out easier to read and work with (and less likely to cause merge conflicts).